src/main/java/com/datasonnet/Regex.java

```diff
@@ -0,0 +1,119 @@
+                nextFindNode.add(matcher.group(i));
+            }
+
+            Map<String, Integer> namedGroups = getNamedGroupsFromMatcher(matcher);
+            if (!namedGroups.isEmpty()) {
+                ObjectNode nextCaptureGroup = namedCapturesNode.addObject();
+                for (Map.Entry<String, Integer> namedGroup : namedGroups.entrySet()) {
+                    nextCaptureGroup.put(namedGroup.getKey(), matcher.group(namedGroup.getValue()));
+                }
+            }
+        } while (matcher.find());
+
+        regexMatch.set("captures", capturesNode);
+        regexMatch.set("namedCaptures", namedCapturesNode);
+
+        return UjsonUtil.jsonObjectValueOf(regexMatch.toString());
+    }
+
+    public static String regexQuoteMeta(String str) {
+        return Pattern.quote(str);
+    }
+
+    public static String regexReplace(String str, String pattern, String replace) {
+        return replace(str, pattern, replace, false);
+    }
+
+    public static String regexGlobalReplace(String str, String pattern, String replace) {
+        return replace(str, pattern, replace, true);
+    }
+
+    private static String replace(String str, String pattern, String replace, boolean isGlobal) {
+        Matcher matcher = Pattern.compile(pattern).matcher(str);
+        return isGlobal ? matcher.replaceAll(replace) : matcher.replaceFirst(replace);
+    }
+
+    private static Value regexMatch(String expr, String str, boolean isFull) {
+        Pattern pattern = Pattern.compile(expr);
+        Matcher matcher = pattern.matcher(str);
+
+        boolean hasMatch = isFull ? matcher.matches() : matcher.find();
+        if (!hasMatch) {
+            return Value.Null();
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode regexMatch = mapper.createObjectNode();
+        regexMatch.put("string", str);
+
+        ArrayNode capturesNode = mapper.createArrayNode();
+        for (int i = 1; i <= matcher.groupCount(); i++) {
+            capturesNode.add(matcher.group(i));
+        }
+        regexMatch.set("captures", capturesNode);
+
+        ObjectNode namedCapturesNode = mapper.createObjectNode();
+        Map<String, Integer> namedGroups = getNamedGroupsFromMatcher(matcher);
+        for (Map.Entry<String, Integer> namedGroup : namedGroups.entrySet()) {
+            namedCapturesNode.put(namedGroup.getKey(), matcher.group(namedGroup.getValue()));
+        }
+
+        regexMatch.set("namedCaptures", namedCapturesNode);
+
+        return UjsonUtil.jsonObjectValueOf(regexMatch.toString());
+    }
+
+    private static Map<String, Integer> getNamedGroupsFromMatcher(Matcher matcher) {
+        try {
+            Field namedGroupsMapField = Matcher.class.getDeclaredField("namedGroups");
+            namedGroupsMapField.setAccessible(true);
+            return (Map<String, Integer>) namedGroupsMapField.get(matcher);
+        } catch (Exception e) {
+            //TODO log the error?
```

Comment:
This is still silently swallowing the error. It needs to propagate (probably wrapped up). It should also be a more specific error/errors.

Reply:
Well, the only thing that can throw an exception (java.lang.NoSuchFieldException) here is the `getDeclaredField` invocation - which should never throw an exception if the Matcher class is in the classpath (which is always the case). So I really don't think there's a need to propagate this exception.

Reply:
If we don’t, we will never know if there is an exception. We’ll just get mysterious but reports with no clear symptoms at some future time, likely because someone is using an unusual class loader setup

Reply:
OK, it throws exception now.